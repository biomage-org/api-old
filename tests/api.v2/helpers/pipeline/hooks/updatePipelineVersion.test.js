const updatePipelineVersion = require('../../../../../src/api.v2/helpers/pipeline/hooks/updatePipelineVersion');
const Experiment = require('../../../../../src/api.v2/model/Experiment');

jest.mock('../../../../../src/api.v2/model/Experiment');

const message = {
  experimentId: 'mockexp',
  input: {
    authJWT: 'somejwt',
    processName: 'gem2s',
    taskName: 'task1',
  },
  pipelineVersion: 2,
};

const experimentInstance = new Experiment();

describe('updatePipelineVersion', () => {
  it('Works correctly', async () => {
    experimentInstance.updateById.mockImplementationOnce(() => Promise.resolve({}));

    await updatePipelineVersion(message);

    expect(experimentInstance.updateById).toHaveBeenCalledTimes(1);
    expect(experimentInstance.updateById.mock.calls).toMatchSnapshot();
  });
});
