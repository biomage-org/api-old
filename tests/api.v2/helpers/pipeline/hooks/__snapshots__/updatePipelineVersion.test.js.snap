// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`updatePipelineVersion Works correctly 1`] = `
Array [
  Array [
    "mockexp",
    Object {
      "pipeline_version": 2,
    },
  ],
]
`;
